src/aerie_cli/aerie_client.py

```diff
@@ -1361,30 +1404,65 @@ def get_typescript_dictionary(self, command_dictionary_id: int) -> str:
         )
 
         return typescript_dictionary_string
-    
-    def upload_scheduling_goal(self, model_id, name, definition):
-        upload_scheduling_goal_query = """
-        mutation InsertGoal($model_id: Int!, 
-                            $name: String!, 
-                            $definition: String!) {
-            insert_scheduling_goal_one(object: 
-                {
-                model_id: $model_id,
-                name: $name,
-                definition: $definition,
-                }) {
+
+    def get_scheduling_goals_by_specification(self, spec_id):
+        list_all_goals_by_spec_query = """
+        query ($spec: Int!){
+            scheduling_specification_goals(where: {
+                specification_id:{_eq:$spec}
+            }){
+                goal{
                     id
+                    model_id
+                    name
+                    description
+                    author
+                    last_modified_by
+                    created_date
+                    modified_date
                 }
-        }"""
+            }
+        }
+        """
+        resp = self.host_session.post_to_graphql(list_all_goals_by_spec_query, spec=spec_id)
+
+        return resp
 
+    def upload_scheduling_goal(self, model_id, name, definition):
+        obj = dict()
+        obj["name"] = name
+        obj["model_id"] = model_id
+        obj["definition"] = definition
+
+        return self.upload_scheduling_goals(obj)
```

Comment:
```suggestion
        return self.upload_scheduling_goals([obj])
```

I think `obj` needs to be put into a list here